learn/cookbooks/multitenancy_nodejs.mdx

```diff
@@ -0,0 +1,214 @@
+
+In Meilisearch, you might have one index containing data belonging to many distinct tenants. In such cases, your tenants must only be able to search through their own documents. You can implement this using [tenant tokens](/learn/security/tenant_tokens).
+
+## Requirements
+
+For this guide, you will need:
+- [Node.js](https://nodejs.org/en) and a package manager like `npm`, `yarn`, or `pnpm`
+- [Meilisearch JavaScript SDK](/learn/what_is_meilisearch/sdks)
+- A Meilisearch server running — see our [quick start](/learn/getting_started/cloud_quick_start)
+- A search API key — available in your Meilisearch dashboard
+- A search API key UID — retrieve it using the [keys endpoints](/reference/api/keys)
+
+<Capsule intent="tip">
+Prefer self-hosting? Read our [installation guide](/learn/getting_started/installation).
+</Capsule>
+
+## Data models
+
+In this guide, we'll use a simple data model to represent medical appointments. The documents in the Meilisearch index will look like this:
+
+```json
+[
+  {
+    "id": 1,
+    "patient": "John",
+    "details": "I think I caught a cold. Can you help me?",
+    "status": "pending"
+  },
+  {
+    "id": 2,
+    "patient": "Zia",
+    "details": "I'm suffering from fever. I need an appointment ASAP.",
+    "status": "pending"
+  },
+  {
+    "id": 3,
+    "patient": "Kevin",
+    "details": "Some confidential information Kevin has shared.",
+    "status": "confirmed"
+  }
+]
+```
+
+For the purpose of this guide, we assume documents are stored in an `appointments` index.
+
+## Creating a tenant token
+
+You will generate a tenant token that will allow a given patient to search only for their appointments. To achieve this, you will create a tenant token that filters the results based on the patient's ID.
+
+In a `search.js` file, you can create the tenant token like this:
+
+```js
+// search.js
+
+import { Meilisearch } from 'meilisearch'
+
+const apiKey = '<YOUR_SEARCH_API_KEY>'
+const apiKeyUid = '<YOUR_SEARCH_API_KEY_UID>'
+const indexName = 'appointments'
+
+const client = new Meilisearch({
+	host: 'https://edge.meilisearch.com', // Your Meilisearch host
+	apiKey: apiKey
+})
+
+export function createTenantToken(patientName) {
+  const searchRules = {
+    [indexName]: {
+      'filter': `patient = ${patientName}`
+    }
+  }
+
+  const tenantToken = client.generateTenantToken(
+    apiKeyUid,
+    searchRules,
+    {
+      expiresAt: new Date('2030-01-01'), // Choose an expiration date
+      apiKey: apiKey,
+    }
+  )
+  return tenantToken
+}
+```
+
+When Meilisearch gets a search query with a tenant token, it decodes it and applies the search rules to the search request. In this example, the results are filtered by the `patient` field. This means that a patient can only search for their own appointments.
+
+Next, you will use this tenant token to perform searches against your Meilisearch instance.
```

Comment:
I think you don't need this sentence, since the first paragraph in the next section more or less repeats the same thing. Your call, though, feel free to ignore if you disagree.

```suggestion
```